geospatial/src/Dem.cc

```diff
@@ -103,46 +97,55 @@ int Dem::Load(const std::string &_filename)
 
   if (this->dataPtr->dataSet == nullptr)
   {
-    gzerr << "Unable to open DEM file[" << fullName
+    ignerr << "Unable to open DEM file[" << fullName
           << "]. Format not recognised as a supported dataset." << std::endl;
     return -1;
   }
 
-  int nBands = this->dataPtr->dataSet->RasterCount();
+  int nBands = this->dataPtr->dataSet->GetRasterCount();
   if (nBands != 1)
   {
-    gzerr << "Unsupported number of bands in file [" << fullName + "]. Found "
+    ignerr << "Unsupported number of bands in file [" << fullName + "]. Found "
           << nBands << " but only 1 is a valid value." << std::endl;
     return -1;
   }
 
   // Set the pointer to the band
-  this->dataPtr->band = this->dataPtr->dataSet->RasterBand(1);
+  this->dataPtr->band = this->dataPtr->dataSet->GetRasterBand(1);
 
   // Raster width and height
-  xSize = this->dataPtr->dataSet->RasterXSize();
-  ySize = this->dataPtr->dataSet->RasterYSize();
+  xSize = this->dataPtr->dataSet->GetRasterXSize();
+  ySize = this->dataPtr->dataSet->GetRasterYSize();
 
   // Corner coordinates
-  upLeftX = 0.0;
-  upLeftY = 0.0;
-  upRightX = xSize;
-  upRightY = 0.0;
-  lowLeftX = 0.0;
-  lowLeftY = ySize;
-
-  // Calculate the georeferenced coordinates of the terrain corners
-  this->GeoReference(upLeftX, upLeftY, upLeftLat, upLeftLong);
-  this->GeoReference(upRightX, upRightY, upRightLat, upRightLong);
-  this->GeoReference(lowLeftX, lowLeftY, lowLeftLat, lowLeftLong);
-
-  // Set the world width and height
-  this->dataPtr->worldWidth =
-     math::SphericalCoordinates::Distance(upLeftLat, upLeftLong,
-                                            upRightLat, upRightLong);
-  this->dataPtr->worldHeight =
-     math::SphericalCoordinates::Distance(upLeftLat, upLeftLong,
-                                            lowLeftLat, lowLeftLong);
+  try
+  {
```

Comment:
I know this was ported straight from Gazebo classic, but can this try block be narrower, including just the lines that throw an exception?

Reply:
I've gone ahead and removed the `try/catch` block and replaced it with an error instead. Also, instead of continuing through the code, it returns when the error is reached 6f6769aedb2ca1689d3a08f295036edec461c099

Reply:
yeah we had to throw an exception in gazebo11 since the `GeoReference` method returned `void` and its API was already frozen. I notice that you changed that API to return `bool` in this PR, which is much better. 👍